File: api/v1.0.4/src/Entity/Usuario.php
<?php
namespace src\Entity;

use src\Entity\Contract\Entity;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\ORM\Mapping AS ORM;
use JMS\Serializer\Annotation as JMS;


/**
 * @ORM\Table("usuarios")
 * @ORM\Entity
 */
class Usuario implements Entity
{
    /**
     * @JMS\Groups({"list"})
     *
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="AUTO")
     * @ORM\Column(name="id", type="integer")
     */private $id;

    /**
     * @ORM\Column(type="string")
     */private $bairro;
    /**
     * @ORM\Column(type="string")
     */private $cargo;
    /**
     * @ORM\Column(type="string")
     */private $celular;
    /**
     * @ORM\Column(type="string")
     */private $cEP;
    /**
     * @ORM\Column(type="string")
     */private $cidade;
    /**
     * @ORM\Column(type="string")
     */private $cNPJMEI;
    /**
     * @ORM\Column(type="string")
     */private $complemento;
    /**
     * @ORM\Column(type="string")
     */private $cPF;
    /**
     * @ORM\Column(type="string")
     */private $curriculoResumido;
    /**
     * @ORM\Column(type="string")
     */private $dataDeAdmissao;
    /**
     * @ORM\Column(type="string")
     */private $DataDeDesligamento;
    /**
     * @ORM\Column(type="string")
     */private $dataDeExpedicao;
    /**
     * @ORM\Column(type="string")
     */private $dataDeNascimento;
    /**
     * @ORM\Column(type="string")
     */private $dataExpedicaoCTPS;
    /**
     * @ORM\Column(type="string")
     */private $disciplina;
    /**
     * @ORM\Column(type="string")
     */private $email;
    /**
     * @ORM\Column(type="string")
     */private $endereco;
    /**
     * @ORM\Column(type="string")
     */private $estado;
    /**
     * @ORM\Column(type="string")
     */private $estadoCivil;
    /**
     * @ORM\Column(type="string")
     */private $formacao;
    /**
     * @ORM\Column(type="string")
     */private $habilitacao;
    /**
     * @ORM\Column(type="string")
     */private $nome;
    /**
     * @ORM\Column(type="string")
     */private $numero;
    /**
     * @ORM\Column(type="string")
     */private $numeroCTPS;
    /**
     * @ORM\Column(type="string")
     */private $observacoes;
    /**
     * @ORM\Column(type="string")
     */private $orgaoEmissor;
    /**
     * @ORM\Column(type="integer")
     */private $permissoes_id;
    /**
     * @ORM\Column(type="string")
     */private $regiao;
    /**
     * @ORM\Column(type="string")
     */private $relacaoFuncional;
    /**
     * @ORM\Column(type="string")
     */private $rG;
    /**
     * @ORM\Column(type="string")
     */private $serieCTPS;
    /**
     * @ORM\Column(type="string")
     */private $sexo;
    /**
     * @ORM\Column(type="string")
     */private $status;
    /**
     * @ORM\Column(type="string")
     */private $telefone;
    /**
     * @ORM\Column(type="string")
     */private $uFCTPS;

    /**
     * @return mixed
     */
    public function getId()
    {
        return $this->id;
    }

    /**
     * @param mixed $id
     */
    public function setId($id)
    {
        $this->id = $id;
    }

    /**
     * @return mixed
     */
    public function getBairro()
    {
        return $this->bairro;
    }

    /**
     * @param mixed $bairro
     */
    public function setBairro($bairro)
    {
        $this->bairro = $bairro;
    }

    /**
     * @return mixed
     */
    public function getCargo()
    {
        return $this->cargo;
    }

    /**
     * @param mixed $cargo
     */
    public function setCargo($cargo)
    {
        $this->cargo = $cargo;
    }

    /**
     * @return mixed
     */
    public function getCelular()
    {
        return $this->celular;
    }

    /**
     * @param mixed $celular
     */
    public function setCelular($celular)
    {
        $this->celular = $celular;
    }

    /**
     * @return mixed
     */
    public function getCEP()
    {
        return $this->cEP;
    }

    /**
     * @param mixed $cEP
     */
    public function setCEP($cEP)
    {
        $this->cEP = $cEP;
    }

    /**
     * @return mixed
     */
    public function getCidade()
    {
        return $this->cidade;
    }

    /**
     * @param mixed $cidade
     */
    public function setCidade($cidade)
    {
        $this->cidade = $cidade;
    }

    /**
     * @return mixed
     */
    public function getCNPJMEI()
    {
        return $this->cNPJMEI;
    }

    /**
     * @param mixed $cNPJMEI
     */
    public function setCNPJMEI($cNPJMEI)
    {
        $this->cNPJMEI = $cNPJMEI;
    }

    /**
     * @return mixed
     */
    public function getComplemento()
    {
        return $this->complemento;
    }

    /**
     * @param mixed $complemento
     */
    public function setComplemento($complemento)
    {
        $this->complemento = $complemento;
    }

    /**
     * @return mixed
     */
    public function getCPF()
    {
        return $this->cPF;
    }

    /**
     * @param mixed $cPF
     */
    public function setCPF($cPF)
    {
        $this->cPF = $cPF;
    }

    /**
     * @return mixed
     */
    public function getCurriculoResumido()
    {
        return $this->curriculoResumido;
    }

    /**
     * @param mixed $curriculoResumido
     */
    public function setCurriculoResumido($curriculoResumido)
    {
        $this->curriculoResumido = $curriculoResumido;
    }

    /**
     * @return mixed
     */
    public function getDataDeAdmissao()
    {
        return $this->dataDeAdmissao;
    }

    /**
     * @param mixed $dataDeAdmissao
     */
    public function setDataDeAdmissao($dataDeAdmissao)
    {
        $this->dataDeAdmissao = $dataDeAdmissao;
    }

    /**
     * @return mixed
     */
    public function getDataDeDesligamento()
    {
        return $this->DataDeDesligamento;
    }

    /**
     * @param mixed $DataDeDesligamento
     */
    public function setDataDeDesligamento($DataDeDesligamento)
    {
        $this->DataDeDesligamento = $DataDeDesligamento;
    }

    /**
     * @return mixed
     */
    public function getDataDeExpedicao()
    {
        return $this->dataDeExpedicao;
    }

    /**
     * @param mixed $dataDeExpedicao
     */
    public function setDataDeExpedicao($dataDeExpedicao)
    {
        $this->dataDeExpedicao = $dataDeExpedicao;
    }

    /**
     * @return mixed
     */
    public function getDataDeNascimento()
    {
        return $this->dataDeNascimento;
    }

    /**
     * @param mixed $dataDeNascimento
     */
    public function setDataDeNascimento($dataDeNascimento)
    {
        $this->dataDeNascimento = $dataDeNascimento;
    }

    /**
     * @return mixed
     */
    public function getDataExpedicaoCTPS()
    {
        return $this->dataExpedicaoCTPS;
    }

    /**
     * @param mixed $dataExpedicaoCTPS
     */
    public function setDataExpedicaoCTPS($dataExpedicaoCTPS)
    {
        $this->dataExpedicaoCTPS = $dataExpedicaoCTPS;
    }

    /**
     * @return mixed
     */
    public function getDisciplina()
    {
        return $this->disciplina;
    }

    /**
     * @param mixed $disciplina
     */
    public function setDisciplina($disciplina)
    {
        $this->disciplina = $disciplina;
    }

    /**
     * @return mixed
     */
    public function getEmail()
    {
        return $this->email;
    }

    /**
     * @param mixed $email
     */
    public function setEmail($email)
    {
        $this->email = $email;
    }

    /**
     * @return mixed
     */
    public function getEndereco()
    {
        return $this->endereco;
    }

    /**
     * @param mixed $endereco
     */
    public function setEndereco($endereco)
    {
        $this->endereco = $endereco;
    }

    /**
     * @return mixed
     */
    public function getEstado()
    {
        return $this->estado;
    }

    /**
     * @param mixed $estado
     */
    public function setEstado($estado)
    {
        $this->estado = $estado;
    }

    /**
     * @return mixed
     */
    public function getEstadoCivil()
    {
        return $this->estadoCivil;
    }

    /**
     * @param mixed $estadoCivil
     */
    public function setEstadoCivil($estadoCivil)
    {
        $this->estadoCivil = $estadoCivil;
    }

    /**
     * @return mixed
     */
    public function getFormacao()
    {
        return $this->formacao;
    }

    /**
     * @param mixed $formacao
     */
    public function setFormacao($formacao)
    {
        $this->formacao = $formacao;
    }

    /**
     * @return mixed
     */
    public function getHabilitacao()
    {
        return $this->habilitacao;
    }

    /**
     * @param mixed $habilitacao
     */
    public function setHabilitacao($habilitacao)
    {
        $this->habilitacao = $habilitacao;
    }

    /**
     * @return mixed
     */
    public function getNome()
    {
        return $this->nome;
    }

    /**
     * @param mixed $nome
     */
    public function setNome($nome)
    {
        $this->nome = $nome;
    }

    /**
     * @return mixed
     */
    public function getNumero()
    {
        return $this->numero;
    }

    /**
     * @param mixed $numero
     */
    public function setNumero($numero)
    {
        $this->numero = $numero;
    }

    /**
     * @return mixed
     */
    public function getNumeroCTPS()
    {
        return $this->numeroCTPS;
    }

    /**
     * @param mixed $numeroCTPS
     */
    public function setNumeroCTPS($numeroCTPS)
    {
        $this->numeroCTPS = $numeroCTPS;
    }

    /**
     * @return mixed
     */
    public function getObservacoes()
    {
        return $this->observacoes;
    }

    /**
     * @param mixed $observacoes
     */
    public function setObservacoes($observacoes)
    {
        $this->observacoes = $observacoes;
    }

    /**
     * @return mixed
     */
    public function getOrgaoEmissor()
    {
        return $this->orgaoEmissor;
    }

    /**
     * @param mixed $orgaoEmissor
     */
    public function setOrgaoEmissor($orgaoEmissor)
    {
        $this->orgaoEmissor = $orgaoEmissor;
    }

    /**
     * @return mixed
     */
    public function getPermissoesId()
    {
        return $this->permissoes_id;
    }

    /**
     * @param mixed $permissoes_id
     */
    public function setPermissoesId($permissoes_id)
    {
        $this->permissoes_id = $permissoes_id;
    }

    /**
     * @return mixed
     */
    public function getRegiao()
    {
        return $this->regiao;
    }

    /**
     * @param mixed $regiao
     */
    public function setRegiao($regiao)
    {
        $this->regiao = $regiao;
    }

    /**
     * @return mixed
     */
    public function getRelacaoFuncional()
    {
        return $this->relacaoFuncional;
    }

    /**
     * @param mixed $relacaoFuncional
     */
    public function setRelacaoFuncional($relacaoFuncional)
    {
        $this->relacaoFuncional = $relacaoFuncional;
    }

    /**
     * @return mixed
     */
    public function getRG()
    {
        return $this->rG;
    }

    /**
     * @param mixed $rG
     */
    public function setRG($rG)
    {
        $this->rG = $rG;
    }

    /**
     * @return mixed
     */
    public function getSerieCTPS()
    {
        return $this->serieCTPS;
    }

    /**
     * @param mixed $serieCTPS
     */
    public function setSerieCTPS($serieCTPS)
    {
        $this->serieCTPS = $serieCTPS;
    }

    /**
     * @return mixed
     */
    public function getSexo()
    {
        return $this->sexo;
    }

    /**
     * @param mixed $sexo
     */
    public function setSexo($sexo)
    {
        $this->sexo = $sexo;
    }

    /**
     * @return mixed
     */
    public function getStatus()
    {
        return $this->status;
    }

    /**
     * @param mixed $status
     */
    public function setStatus($status)
    {
        $this->status = $status;
    }

    /**
     * @return mixed
     */
    public function getTelefone()
    {
        return $this->telefone;
    }

    /**
     * @param mixed $telefone
     */
    public function setTelefone($telefone)
    {
        $this->telefone = $telefone;
    }

    /**
     * @return mixed
     */
    public function getUFCTPS()
    {
        return $this->uFCTPS;
    }

    /**
     * @param mixed $uFCTPS
     */
    public function setUFCTPS($uFCTPS)
    {
        $this->uFCTPS = $uFCTPS;
    }





}

File: api/JAVA/workspace/jpa/src/main/java/br/net/ipp/jpa/model/configuracoes/Usuario.java
package br.net.ipp.jpa.model.configuracoes;

import java.util.Date;

import javax.persistence.Entity;
import javax.persistence.ManyToOne;

import org.hibernate.validator.constraints.NotEmpty;

import br.net.ipp.jpa.model.User;
import br.net.ipp.jpa.model.enums.EstadoCivil;
import br.net.ipp.jpa.model.enums.Regiao;
import br.net.ipp.jpa.model.enums.RelacaoFuncional;
import br.net.ipp.jpa.model.enums.Sexo;
import br.net.ipp.jpa.model.enums.Status;

@Entity
public class Usuario extends User {

	private static final long serialVersionUID = 1L;

	@NotEmpty
	private String nome;
	private String telefone;
	private String celular;
	private Sexo sexo;
	private Status status;

	private String endereco;
	private String numero;
	private String complemento;
	private String bairro;
	private String cidade;
	private String estado;
	private String cEP;
	private Regiao regiao;

	private String numeroCTPS;
	private String serieCTPS;
	private String orgaoEmissor;
	private String uFCTPS;
	private Date dataExpedicaoCTPS;

	private Date dataDeNascimento;
	private EstadoCivil estadoCivil;
	private String formacao;

	private String habilitacao;
	private String dataDeExpedicao;

	private String cNPJMEI;
	private int cPF;
	private String rG;

	private String curriculoResumido;
	private Date dataDeAdmissao;
	private Date DataDeDesligamento;
	private String disciplina;
	private String cargo;
	private RelacaoFuncional relacaoFuncional;

	@ManyToOne
	private GrupoDePermissoes grupoDePermissoes;
	private String observacoes;

	public String getNome() {
		return nome;
	}

	public void setNome(String nome) {
		this.nome = nome;
	}

	public String getTelefone() {
		return telefone;
	}

	public void setTelefone(String telefone) {
		this.telefone = telefone;
	}

	public String getCelular() {
		return celular;
	}

	public void setCelular(String celular) {
		this.celular = celular;
	}

	public Sexo getSexo() {
		return sexo;
	}

	public void setSexo(Sexo sexo) {
		this.sexo = sexo;
	}

	public Status getStatus() {
		return status;
	}

	public void setStatus(Status status) {
		this.status = status;
	}

	public String getEndereco() {
		return endereco;
	}

	public void setEndereco(String endereco) {
		this.endereco = endereco;
	}

	public String getNumero() {
		return numero;
	}

	public void setNumero(String numero) {
		this.numero = numero;
	}

	public String getComplemento() {
		return complemento;
	}

	public void setComplemento(String complemento) {
		this.complemento = complemento;
	}

	public String getBairro() {
		return bairro;
	}

	public void setBairro(String bairro) {
		this.bairro = bairro;
	}

	public String getCidade() {
		return cidade;
	}

	public void setCidade(String cidade) {
		this.cidade = cidade;
	}

	public String getEstado() {
		return estado;
	}

	public void setEstado(String estado) {
		this.estado = estado;
	}

	public String getcEP() {
		return cEP;
	}

	public void setcEP(String cEP) {
		this.cEP = cEP;
	}

	public Regiao getRegiao() {
		return regiao;
	}

	public void setRegiao(Regiao regiao) {
		this.regiao = regiao;
	}

	public String getNumeroCTPS() {
		return numeroCTPS;
	}

	public void setNumeroCTPS(String numeroCTPS) {
		this.numeroCTPS = numeroCTPS;
	}

	public String getSerieCTPS() {
		return serieCTPS;
	}

	public void setSerieCTPS(String serieCTPS) {
		this.serieCTPS = serieCTPS;
	}

	public String getOrgaoEmissor() {
		return orgaoEmissor;
	}

	public void setOrgaoEmissor(String orgaoEmissor) {
		this.orgaoEmissor = orgaoEmissor;
	}

	public String getuFCTPS() {
		return uFCTPS;
	}

	public void setuFCTPS(String uFCTPS) {
		this.uFCTPS = uFCTPS;
	}

	public Date getDataExpedicaoCTPS() {
		return dataExpedicaoCTPS;
	}

	public void setDataExpedicaoCTPS(Date dataExpedicaoCTPS) {
		this.dataExpedicaoCTPS = dataExpedicaoCTPS;
	}

	public Date getDataDeNascimento() {
		return dataDeNascimento;
	}

	public void setDataDeNascimento(Date dataDeNascimento) {
		this.dataDeNascimento = dataDeNascimento;
	}

	public EstadoCivil getEstadoCivil() {
		return estadoCivil;
	}

	public void setEstadoCivil(EstadoCivil estadoCivil) {
		this.estadoCivil = estadoCivil;
	}

	public String getFormacao() {
		return formacao;
	}

	public void setFormacao(String formacao) {
		this.formacao = formacao;
	}

	public String getHabilitacao() {
		return habilitacao;
	}

	public void setHabilitacao(String habilitacao) {
		this.habilitacao = habilitacao;
	}

	public String getDataDeExpedicao() {
		return dataDeExpedicao;
	}

	public void setDataDeExpedicao(String dataDeExpedicao) {
		this.dataDeExpedicao = dataDeExpedicao;
	}

	public String getcNPJMEI() {
		return cNPJMEI;
	}

	public void setcNPJMEI(String cNPJMEI) {
		this.cNPJMEI = cNPJMEI;
	}

	public int getcPF() {
		return cPF;
	}

	public void setcPF(int cPF) {
		this.cPF = cPF;
	}

	public String getrG() {
		return rG;
	}

	public void setrG(String rG) {
		this.rG = rG;
	}

	public String getCurriculoResumido() {
		return curriculoResumido;
	}

	public void setCurriculoResumido(String curriculoResumido) {
		this.curriculoResumido = curriculoResumido;
	}

	public Date getDataDeAdmissao() {
		return dataDeAdmissao;
	}

	public void setDataDeAdmissao(Date dataDeAdmissao) {
		this.dataDeAdmissao = dataDeAdmissao;
	}

	public Date getDataDeDesligamento() {
		return DataDeDesligamento;
	}

	public void setDataDeDesligamento(Date dataDeDesligamento) {
		DataDeDesligamento = dataDeDesligamento;
	}

	public String getDisciplina() {
		return disciplina;
	}

	public void setDisciplina(String disciplina) {
		this.disciplina = disciplina;
	}

	public String getCargo() {
		return cargo;
	}

	public void setCargo(String cargo) {
		this.cargo = cargo;
	}

	public RelacaoFuncional getRelacaoFuncional() {
		return relacaoFuncional;
	}

	public void setRelacaoFuncional(RelacaoFuncional relacaoFuncional) {
		this.relacaoFuncional = relacaoFuncional;
	}

	public GrupoDePermissoes getGrupoDePermissoes() {
		return grupoDePermissoes;
	}

	public void setGrupoDePermissoes(GrupoDePermissoes grupoDePermissoes) {
		this.grupoDePermissoes = grupoDePermissoes;
	}

	public String getObservacoes() {
		return observacoes;
	}

	public void setObservacoes(String observacoes) {
		this.observacoes = observacoes;
	}

}
